core/src/main/java/bisq/core/xmr/knaccc/monero/address/WalletAddress.java

```diff
@@ -131,7 +131,13 @@ public String getSubaddressBase58(String privateViewKeyHex, long accountId, long
     }
 
     public boolean checkPrivateViewKey(String privateViewKey) {
-        return arePubPrivKeysRelated(this.publicViewKeyHex, privateViewKey);
+        return isPrivateKeyValid(privateViewKey) && arePubPrivKeysRelated(this.publicViewKeyHex, privateViewKey);
+    }
+
+    public static boolean isPrivateKeyValid(String privateKey) {
```

Comment:
Add @VisibleForTesting and make package scope after moving test in correct package